### PR TITLE
beamPackages.hex: 0.20.1 -> 0.20.5

### PR DESCRIPTION
**Pull request by cw789**
###### Motivation for this change
 Keeping things up-to-date.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cw789** [comment]
cc @ericbmerritt

**Comment by cw789** [comment]
Calling the Elixir maintainers:
@the-kenny @havvy @couchemar @ankhers @filalex77

Hey guys, could anybody help me to get this merged?

**Review by Br1ght0ne — APPROVED** [review]
diff LGTM

**Comment by filalex77** [comment]
@cw789 Sadly, I don't have commit access. Pinging people who have :)

cc @marsam @yegortimoshenko. Thanks a lot!

**Review by lukateras — APPROVED** [review]

File: pkgs/development/beam-modules/hex/default.nix [hunk 1]
```diff
@@ -8,13 +8,15 @@ let
 
   pkg = self: stdenv.mkDerivation rec {
     pname = "hex";
-    version = "0.20.1";
+    version = "0.20.5";
 
+    # How to obtain `sha256`:
+    # nix-prefetch-url --unpack https://github.com/hexpm/hex/archive/v${version}.tar.gz
```
Comment (yegortimoshenko) [inline pkgs/development/beam-modules/hex/default.nix]: Frankly, that's true of every single `fetchFromGitHub` case, and I'm not sure this comment is deserved here. (Or one could also use `nix-prefetch-github`).

**Comment by yegortimoshenko** [comment]
```
installing
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/c9bhvsszng7fawgmj2cn3fq12zr320y3-hex-0.20.5
patching script interpreter paths in /nix/store/c9bhvsszng7fawgmj2cn3fq12zr320y3-hex-0.20.5
checking for references to /build/ in /nix/store/c9bhvsszng7fawgmj2cn3fq12zr320y3-hex-0.20.5...
/nix/store/c9bhvsszng7fawgmj2cn3fq12zr320y3-hex-0.20.5
```

**Comment by yegortimoshenko** [comment]
@GrahamcOfBorg build beamPackages.hex beamPackages.pc

File: pkgs/development/beam-modules/hex/default.nix (hunk 1 above)
Comment (cw789) [reply pkgs/development/beam-modules/hex/default.nix]: I wasn't aware of `nix-prefetch-github`.
But I prefer to communicate the universal usable method of `nix-prefetch-url`.

I've started to add this comment on nixpkgs, as I still find it not easy for new peoples to find out the easy way of updating a pkg with getting the `sha256`....

Comment (cw789) [reply pkgs/development/beam-modules/hex/default.nix]: But of course, I can remove it here.
But I still would add this on pkgs where the fetchFromGitHub is not do be found directly beneath the `version`.

Comment (yegortimoshenko) [reply pkgs/development/beam-modules/hex/default.nix]:
```
$ nix-shell -p nix-prefetch-github --run 'nix-prefetch-github hexpm hex --rev v0.20.5'
{
    "owner": "hexpm",
    "repo": "hex",
    "rev": "3f49ed54d2858f6fdb9a06c3f7b242b3dbe1d081",
    "sha256": "1wz6n4qrmsb4kkww6lrdbs99xzwp4dyjjmr8m4drcwn3sd2k9ba6"
}
```

Comment (cw789) [reply pkgs/development/beam-modules/hex/default.nix]: Thanks a lot for removing it.
I would think of some ways to communicate this in future more easily to new contributors...

**Pull request merged**